src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewAction.java

```diff
@@ -67,12 +63,24 @@ public void getSteps(StaplerRequest req, StaplerResponse rsp) throws IOException
       rsp.getWriter().append(stepsJson);
     } else {
       logger.debug("getSteps was not passed nodeId.");
-      // Consider returning the full map in one go here - the frontend will need to be updated to
-      // handle this.
       rsp.getWriter().append("Error getting console text");
     }
   }
 
+  // Return all steps to:
+  // - reduce number of API calls
+  // - remove dependency of getting list of stages in frontend.
+  @WebMethod(name = "allSteps")
+  public void getAllSteps(StaplerRequest req, StaplerResponse rsp) throws IOException {
+    ObjectMapper mapper = new ObjectMapper();
+    PipelineStepList steps = stepApi.getAllSteps();
+    String stepsJson = mapper.writeValueAsString(steps);
+    if (logger.isDebugEnabled()) {
+      logger.debug("Steps: '" + stepsJson + "'.");
+    }
+    rsp.getWriter().append(stepsJson);
```

Comment:
Content-Type header should be added to the response

Reply:
Have switched to using HttpResponses.okJSON/errorJSON.